relay_address_generator_range.go

```diff
@@ -0,0 +1,104 @@
+package turn
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pion/transport/vnet"
+)
+
+// RelayAddressGeneratorPortRange can be used to only allocate connections inside a defined port range.
+// Similar to the RelayAddressGeneratorStatic a static ip address can be set.
+type RelayAddressGeneratorPortRange struct {
+	// RelayAddress is the IP returned to the user when the relay is created
+	RelayAddress net.IP
+
+	// MinPort the minimum port to allocate
+	MinPort uint16
+	// MaxPort the maximum (inclusive) port to allocate
+	MaxPort uint16
+
+	// MaxRetries the amount of tries to allocate a random port in the defined range
+	MaxRetries int
+
+	// Rand the random source of numbers
+	Rand      *rand.Rand
+	randMutex sync.Mutex
+
+	// Address is passed to Listen/ListenPacket when creating the Relay
+	Address string
+
+	Net *vnet.Net
+}
+
+// Validate is called on server startup and confirms the RelayAddressGenerator is properly configured
+func (r *RelayAddressGeneratorPortRange) Validate() error {
+	if r.Net == nil {
+		r.Net = vnet.NewNet(nil)
+	}
+
+	if r.Rand == nil {
+		r.Rand = rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
Can I ignore gosec. 
> Use of weak random number generator (math/rand instead of crypto/rand) (gosec)

Using a crypto rand source isn't really critical here?

Reply:
I would check out https://github.com/pion/randutil it doesn't do much, just cuts down on the amount of duplication we need to do.

Reply:
Nice, didn't knew this existed (:.